### PR TITLE
Infrastructure.test_advanced_search_providers fix and web_ui.search updates

### PR DESCRIPTION
**Pull request by mshriver**
{{ pytest: cfme/tests/infrastructure/test_advanced_search_providers.py }}

Purpose or Intent
=================

Fixing navigation, search, and downstream test failures in test_advanced_search_providers.

Inconsistent test results on back-to-back runs locally, with transient navigation failures dominating. Running through PRT to get more data points.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1645ebb0d6223bee641532cdc8bd05b7a521e084:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3b1a0509e919d936c1656892fcafbea42bc44c52:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8cbc9a2c550278c5617f91a45cdd79dc6c85dc33:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

File: cfme/tests/infrastructure/test_advanced_search_providers.py [hunk 1]
```diff
@@ -36,47 +35,63 @@ def hosts_with_vm_count(hosts):
...
+    from fixtures import ssh_client
+    filter_name = getattr(request.function, "filter_name", None)
+    logger.debug('rails_delete_filter: calling rails to delete filter: {}'.format(filter_name))
+    if filter_name:
+        ssh_client.run_rails_command(
+            '"MiqSearch.where(:description => {}).first.delete"'.format(repr(filter_name)))
+    else:
+        logger.warning('rails_delete_filter: failed to get filter_name')
+
+pytestmark = [pytest.mark.usefixtures("setup_cleanup_search"), pytest.mark.tier(3)]
```
Comment (RonnyPfannschmidt) [inline cfme/tests/infrastructure/test_advanced_search_providers.py]: for preventing later errors, please put pytestmark to the top of the file right after imports

File: cfme/tests/infrastructure/test_advanced_search_providers.py [hunk 2]
```diff
@@ -87,84 +102,71 @@ def test_filter_with_user_input_and_cancellation(providers):
...
-    search.save_filter("fill_count(Infrastructure Provider.VMs, >)", filter_name, cancel=True)
+    assert search.save_filter("fill_count(Infrastructure Provider.VMs, >)", filter_name,
+                              cancel=True)
     assert_no_cfme_exception()
-    with pytest.raises(pytest.sel.NoSuchElementException):
-        search.load_filter(filter_name)  # does not exist
+
+    assert search.reset_filter()
+    # Exception depends on system state - Load button will be disabled if there are no saved filters
+    try:
```
Comment (RonnyPfannschmidt) [inline cfme/tests/infrastructure/test_advanced_search_providers.py]: please use with `pytest.raises((FOO,BAR)):`

**Comment by dajoRH** [comment]
Lint report for commit 8da31533ef396bf2566146f7518ab078d6dcd3f9:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit daa4c6e92d464a64c94cf48b8848d9305a5a6287:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by mshriver** [comment]
PRT failures are for known bugs.

**Comment by dajoRH** [comment]
Lint report for commit d85fc71c2230dc6e4ccd10f8c8be056391b7b54e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Pull request merged**